src/V3Task.cpp

```diff
@@ -419,17 +419,22 @@ class TaskVisitor final : public AstNVisitor {
         return newvscp;
     }
     AstVarScope* createVarScope(AstVar* invarp, const string& name) {
-        // We could create under either the ref's scope or the ftask's scope.
-        // It shouldn't matter, as they are only local variables.
-        // We choose to do it under whichever called this function, which results
-        // in more cache locality.
-        AstVar* newvarp = new AstVar(invarp->fileline(), AstVarType::BLOCKTEMP, name, invarp);
-        newvarp->funcLocal(false);
-        newvarp->propagateAttrFrom(invarp);
-        m_modp->addStmtp(newvarp);
-        AstVarScope* newvscp = new AstVarScope(newvarp->fileline(), m_scopep, newvarp);
-        m_scopep->addVarp(newvscp);
-        return newvscp;
+        if (invarp->isParam() && VN_IS(invarp->valuep(), InitArray)) {
+            // Move array params in functions into constant pool
+            return v3Global.rootp()->constPoolp()->findTable(VN_CAST(invarp->valuep(), InitArray));
+        } else {
+            // We could create under either the ref's scope or the ftask's scope.
+            // It shouldn't matter, as they are only local variables.
+            // We choose to do it under whichever called this function, which results
+            // in more cache locality.
+            AstVar* newvarp = new AstVar(invarp->fileline(), AstVarType::BLOCKTEMP, name, invarp);
```

Comment:
For new code please use C++11 bracket init.

```suggestion
            AstVar* newvarp = new AstVar{invarp->fileline(), AstVarType::BLOCKTEMP, name, invarp};
```